### PR TITLE
refactor: go 1.26 modernize

### PR DESCRIPTION
**Pull request by gjermundgaraba**
This is pretty straight-forward: https://go.dev/blog/gofix

## Summary by CodeRabbit

* **Chores**
  * General modernization and internal optimizations across providers, playback, UI rendering, plugins, and tests for improved reliability and maintainability.
  * More consistent parsing and iteration logic for playlists, station lists, lyrics, and metadata to reduce edge-case glitches.
  * Cache persistence updated so playlist timestamp is consistently stored.
  * Minor stability improvements to plugin concurrency and UI scrolling/rendering behavior.

### CODE REVIEW

File: player/icy.go [hunk 1]
```diff
@@ -81,11 +81,11 @@ func (ir *icyReader) consumeMeta() error {
...
 func parseStreamTitle(meta string) string {
 	const prefix = "StreamTitle='"
-	i := strings.Index(meta, prefix)
-	if i < 0 {
+	_, after, ok := strings.Cut(meta, prefix)
+	if !ok {
 		return ""
 	}
-	rest := meta[i+len(prefix):]
+	rest := after
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply player/icy.go]: Fixed
